src/backends/github/API.js

```diff
@@ -229,50 +208,26 @@ export default class API {
   }
 
   composeFileTree(files) {
-    let filename;
-    let part;
-    let parts;
-    let subtree;
-    const fileTree = {};
-
-    files.forEach((file) => {
-      if (file.uploaded) { return; }
-      parts = file.path.split("/").filter(part => part);
-      filename = parts.pop();
-      subtree = fileTree;
-      while (part = parts.shift()) {
-        subtree[part] = subtree[part] || {};
-        subtree = subtree[part];
-      }
-      subtree[filename] = file;
-      file.file = true;
-    });
-
-    return fileTree;
+    return files
+      .map(file => ({ ...file, file: true }))
+      .reduce((tree, file) => tree.setIn(file.path.split("/"), file), Map())
+      .toJS();
   }
 
   persistFiles(entry, mediaFiles, options) {
-    const uploadPromises = [];
-    const files = mediaFiles.concat(entry);
-
-    files.forEach((file) => {
-      if (file.uploaded) { return; }
-      uploadPromises.push(this.uploadBlob(file));
-    });
-
-    const fileTree = this.composeFileTree(files);
+    const newFiles = [...mediaFiles, entry].filter(file => !file.uploaded);
+    const uploadsPromise = Promise.all(newFiles.map(file => this.uploadBlob(file)));
+    const fileTreePromise = uploadsPromise.then(files => this.composeFileTree(files));
+
+    if (options.mode === EDITORIAL_WORKFLOW) {
+      const mediaFilesList = mediaFiles.map(file => ({ path: file.path, sha: file.sha }));
+      return fileTreePromise
+        .then(fileTree => this.editorialWorkflowGit(fileTree, entry, mediaFilesList, options));
+    }
 
-    return Promise.all(uploadPromises).then(() => {
-      if (!options.mode || (options.mode && options.mode === SIMPLE)) {
-        return this.getBranch()
-        .then(branchData => this.updateTree(branchData.commit.sha, "/", fileTree))
-        .then(changeTree => this.commit(options.commitMessage, changeTree))
-        .then(response => this.patchBranch(this.branch, response.sha));
-      } else if (options.mode && options.mode === EDITORIAL_WORKFLOW) {
-        const mediaFilesList = mediaFiles.map(file => ({ path: file.path, sha: file.sha }));
-        return this.editorialWorkflowGit(fileTree, entry, mediaFilesList, options);
-      }
-    });
+    return Promise.all([fileTreePromise, this.getBranch()])
+      .then(([fileTree, branchData]) => this.updateTree(branchData.commit.sha, "/", fileTree))
+      .then(changeTree => this.commit(options.commitMessage, changeTree));
```

Comment:
I don't remember removing this intentionally, so it was likely an oversight. Thanks for the catch!